src/solvers/petsc_nonlinear_solver.C

```diff
@@ -615,15 +642,23 @@ PetscNonlinearSolver<T>::PetscNonlinearSolver (sys_type & system_in) :
   _zero_out_jacobian(true),
   _default_monitor(true),
   _snesmf_reuse_base(true),
-  _computing_base_vector(true)
+  _computing_base_vector(true),
+  _setup_reuse(false),
+  _max_its(0)
 {
 }
 
 
 
 template <typename T>
-PetscNonlinearSolver<T>::~PetscNonlinearSolver () = default;
-
+PetscNonlinearSolver<T>::~PetscNonlinearSolver ()
+{
+  // Take care of the SNES instance, if it was allocated
+  if (_snes)
+    {
+    _snes.destroy();
+    }
+}
```

Comment:
This change should definitely not be necessary, so I'm curious whether some testing you did confirmed that you needed it? The main purpose of the
```
WrappedPetsc<SNES> _snes;
```
object is that it is destroyed automatically when its destructor is called, allowing us to have `default` destructor implementations.

Reply:
I believe it was necessary and I didn't understand why.  I'll check next week -- I'm on conference travel and unfortunately having a great deal of double compiling from scratch on my macbook...

Reply:
Hmm, I really should not be needed.  I'll check when I get back (or if I can get everything to compile without conda while traveling).

Reply:
My tests in MOOSE on actually using the feature pass with the default destructor.  For some reason the tests aren't running now, but once they do this should be good.